### PR TITLE
Add documentation to cirq.decompose protocol regarding specific target gateset

### PR DESCRIPTION
**Pull request by tanujkhattar**
Fixes https://github.com/quantumlib/Cirq/issues/930

### CODE REVIEW

**Review by NoureldinYosri — APPROVED** [review]

File: cirq-core/cirq/protocols/decompose_protocol.py [hunk 1]
```diff
@@ -93,16 +93,12 @@ class SupportsDecompose(Protocol):
...
-    happen to converge towards the X, Y, Z, CZ, PhasedX, specified-matrix gates,
-    and others. This set will vary from release to release. Because of this
-    variability, it is important for consumers of decomposition to look for
-    generic properties of gates, such as "two qubit gate with a unitary matrix",
-    instead of specific gate types such as CZ gates (though a consumer is
-    of course free to handle CZ gates in a special way, and consumers can
-    give an `intercepting_decomposer` to `cirq.decompose` that attempts to
-    target a specific gate set).
+    2-qubit gates included by default in Cirq. If a custom decomposition is not
+    specified, Cirq will decompose all operations to X/Y/Z/CZ/Measurement + Global
```
Comment (dabacon) [inline cirq-core/cirq/protocols/decompose_protocol.py:97]: I'm confused since these aren't universal?  Is it missing PhasedX?

**Comment by dstrain115** [comment]
@tanujkhattar Are we waiting on anything to merge this?

**Review by dstrain115 — APPROVED** [review]

**Comment by dstrain115** [comment]
@tanujkhattar Can we finish this off and commit it?

File: cirq-core/cirq/protocols/decompose_protocol.py (hunk 1 above)
Comment (tanujkhattar) [reply cirq-core/cirq/protocols/decompose_protocol.py]: I meant `XPow/YPow/ZPow/CZPow`; updated the docstring to clarify this.

**Comment by tanujkhattar** [comment]
@dstrain115 This is good to go now.

**Pull request merged**